src/modules/convex/mlp_igd.cpp

```diff
@@ -241,22 +238,19 @@ mlp_minibatch_transition::run(AnyType &args) {
     }
 
     // tuple
-    Matrix indVar;
-    Matrix depVar;
+    MiniBatchTuple tuple;
     try {
         // Ideally there should be no NULLs in the pre-processed input data,
         // but keep it in a try block in case the user has modified the
         // pre-processed data in any way.
-        indVar = args[1].getAs<MappedMatrix>();
-        depVar = args[2].getAs<MappedMatrix>();
+        tuple.indVar = trans(args[1].getAs<MappedMatrix>());
+        tuple.depVar = trans(args[2].getAs<MappedMatrix>());
     } catch (const ArrayWithNullException &e) {
         return args[0];
     }
-    MiniBatchTuple tuple;
+
     // The matrices are by default read as column-major. We will have to
     // transpose it to get back the matrix like how it is in the database.
```

Comment:
Can we move the above two comments to where we initialize `tuple.indVar`?

Reply:
good observation, will do